### PR TITLE
build: Add Dockerfile for building a rhel-atomic image

### PR DESCRIPTION
**Pull request by a-robinson**

### CODE REVIEW

File: build/deploy/rhel/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+MAINTAINER Alex Robinson <alex@cockroachlabs.com>
+
+LABEL name="cockroachdb/cockroach" \
+      vendor="Cockroach Labs" \
+      version="1.0.3" \
+      release="1" \
+      summary="CockroachDB is ..." \
+      description="CockroachDB will ....." \
+      url="https://www.cockroachlabs.com" \
+      io.k8s.description="CockroachDB will do ....." \
```
Comment (tschottdorf) [inline build/deploy/rhel/Dockerfile]: 💯 That'll do, Cockroach. That'll do.

Comment (a-robinson) [inline build/deploy/rhel/Dockerfile]: Haha it'll do whatever it sets its mind to! Updated with descriptions.

**Comment by tamird** [comment]
Let's remember to update release instructions to include updating this file.

**Comment by bdarnell** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---

*[build/deploy/rhel/help.1, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930#-KohScNWHYNMjVjXYxme:-KohScNWHYNMjVjXYxmf:b-u52ich) ([raw file](https://github.com/cockroachdb/cockroach/blob/250000c55da01d811d39124cd3faf52f4f1f07d3/build/deploy/rhel/help.1#L1)):*
> ```Roff
> Documentation for CockroachDB can be found at: https://www.cockroachlabs.com/docs/stable/
> ```

Should this link to 1.0 instead of stable?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930)*

**Comment by a-robinson** [comment]
> Let's remember to update release instructions to include updating this file.

Done, although the process around this is still somewhat TBD as I see how much automation is needed.

---

Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[build/deploy/rhel/help.1, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930#-KohScNWHYNMjVjXYxme:-KohTzwcwApNyHK5GT_S:bwlbm6j) ([raw file](https://github.com/cockroachdb/cockroach/blob/250000c55da01d811d39124cd3faf52f4f1f07d3/build/deploy/rhel/help.1#L1)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Should this link to 1.0 instead of stable?
</blockquote></details>

Yeah, although it'll require us to remember to update this file for every minor release. I'll add that to the release process as well, but all these extra steps add up.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930#-:-KohTlEdq697rXnICKLG:b-4hppv6)*

**Comment by bdarnell** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[build/deploy/rhel/help.1, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930#-KohScNWHYNMjVjXYxme:-KohYXDKNAxGqI_xjd_I:b4gq1zm) ([raw file](https://github.com/cockroachdb/cockroach/blob/250000c55da01d811d39124cd3faf52f4f1f07d3/build/deploy/rhel/help.1#L1)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

Yeah, although it'll require us to remember to update this file for every minor release. I'll add that to the release process as well, but all these extra steps add up.
</blockquote></details>

Only for major releases, not minor ones. At some point we'll also want to generate this file (and all the rest) from templates instead of doing it manually. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930)*

**Comment by a-robinson** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[build/deploy/rhel/help.1, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930#-KohScNWHYNMjVjXYxme:-KohZA6IakyolL2EbkMe:b-zhhfdc) ([raw file](https://github.com/cockroachdb/cockroach/blob/250000c55da01d811d39124cd3faf52f4f1f07d3/build/deploy/rhel/help.1#L1)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

Only for major releases, not minor ones. At some point we'll also want to generate this file (and all the rest) from templates instead of doing it manually. 
</blockquote></details>

In the docs repo, we've already separated out the content for [v1.0](https://github.com/cockroachdb/docs/tree/master/v1.0) from the content for [v1.1](https://github.com/cockroachdb/docs/tree/master/v1.1), so I assume we'll need to update this for 1.1, which is a minor release.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930)*

**Comment by bdarnell** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[build/deploy/rhel/help.1, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930#-KohScNWHYNMjVjXYxme:-KohaMNRkOuY4vuh1x7x:b-pj40d) ([raw file](https://github.com/cockroachdb/cockroach/blob/250000c55da01d811d39124cd3faf52f4f1f07d3/build/deploy/rhel/help.1#L1)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary><blockquote>

In the docs repo, we've already separated out the content for [v1.0](https://github.com/cockroachdb/docs/tree/master/v1.0) from the content for [v1.1](https://github.com/cockroachdb/docs/tree/master/v1.1), so I assume we'll need to update this for 1.1, which is a minor release.
</blockquote></details>

OK. I was interpreting "minor" as e.g. 1.0.3.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930)*

**Comment by a-robinson** [comment]
Review status: 0 of 3 files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---

*[build/deploy/rhel/help.1, line 1 at r2](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930#-KohScNWHYNMjVjXYxme:-KohafVMRUg37ZdClxkO:bi112rj) ([raw file](https://github.com/cockroachdb/cockroach/blob/250000c55da01d811d39124cd3faf52f4f1f07d3/build/deploy/rhel/help.1#L1)):*
<details><summary><i>Previously, bdarnell (Ben Darnell) wrote…</i></summary><blockquote>

OK. I was interpreting "minor" as e.g. 1.0.3.
</blockquote></details>

No worries, that distinction gets me sometimes too. Semver releases are `major.minor.patch`

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/16930)*

**Pull request merged**